### PR TITLE
[MM-23530] Add deployer config file documentation

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
his PR adds documentation for the `deployer` config file.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-23530

### CODE REVIEW

File: docs/deployer_config.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Deployer Configuration
+
+## ClusterName
+
+*string*
+
+Name of the cluster. This will be prefixed to all instances in AWS that built with the configuration.
```
Comment (agnivade) [inline docs/deployer_config.md]: s/instances/resources

that _are_ built

File: docs/deployer_config.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+*integer*
+
+The number of load-test agent instances. The first agent instance will also run the coordinator.
+
+## SSHPublicKey
+
+*string*
+
+The path the SSH public key, this key is used for establishing an SSH connection to the AWS instances.
```
Comment (agnivade) [inline docs/deployer_config.md]: The path to the

File: docs/deployer_config.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+*string*
+
+Type of the Database instance. See types [here](https://aws.amazon.com/rds/instance-types/).
+
+## DBInstanceEngine
+
+*string*
+
+Type of the database backend. This can be either Postgres or MySQL.
```
Comment (agnivade) [inline docs/deployer_config.md]: We can just mention the values here - aurora-mysql or aurora-postgresql

File: docs/deployer_config.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+## ClusterName
+
+*string*
+
+Name of the cluster. This will be prefixed to all instances in AWS that built with the configuration.
+
+## AppInstanceCount
+
+*integer*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*int*
```

File: docs/deployer_config.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+The number of Mattermost application instances.
+
+## AgentCount
+
+*integer*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*int*
```

File: docs/deployer_config.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+## SSHPublicKey
+
+*string*
+
+The path the SSH public key, this key is used for establishing an SSH connection to the AWS instances.
+
+## DBInstanceCount
+
+*integer*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*int*
```

File: docs/deployer_config.md [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+*string*
+
+load-test-ng head reference.
+
+## LogSettings
+
+### EnableConsole
+
+*boolean*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*bool*
```

File: docs/deployer_config.md [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+
+### ConsoleLevel
+
+*string*
+
+Level of detail at which log events are written to the console.
+
+### ConsoleJson
+
+*boolean*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*bool*
```

File: docs/deployer_config.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+
+When true, logged events are written in a machine-readable JSON format. Otherwise, they are printed as plain text.
+
+### EnableFile
+
+*boolean*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*bool*
```

File: docs/deployer_config.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+
+### FileLevel
+
+*string*
+
+Level of detail at which log events are written to log files.
+
+### FileJson
+
+*boolean*
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
*bool*
```

File: docs/deployer_config.md [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+
+*integer*
+
+The number of Mattermost application instances.
+
+## AgentCount
+
+*integer*
+
+The number of load-test agent instances. The first agent instance will also run the coordinator.
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
The number of load-test agent instances. The first instance will also host the [coordinator](coordinator.md).
```

File: docs/deployer_config.md [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+
+*string*
+
+The location of the Mattermost Enterprise Edition license file.
+
+## AdminEmail
+
+*string*
+
+The e-mail will be used when creating a sysadmin user at the deployment process.
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
The e-mail that will be used when creating a sysadmin user during the deployment process.
```

File: docs/deployer_config.md [hunk 13]
```diff
@@ -0,0 +1,141 @@
...
+
+## AdminUsername
+
+*string*
+
+The user name that will be used when creating a sysadmin user at the deployment process.
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
The user name that will be used when creating a sysadmin user during the deployment process.
```

File: docs/deployer_config.md [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+## AdminPassword
+
+*string*
+
+The password that will be used when creating a sysadmin user at the deployment process.
```
Comment (streamer45) [inline docs/deployer_config.md]:
```suggestion
The password that will be used when creating a sysadmin user during the deployment process.
```

File: docs/deployer_config.md [hunk 15]
```diff
@@ -0,0 +1,141 @@
...
+
+*string*
+
+The Go version to download for compiling load-test source.
+
+## SourceCodeRef
+
+*string*
+
+The load-test-ng head reference.
```
Comment (streamer45) [inline docs/deployer_config.md]: Can we expand a bit on this? How is this used? Why is it needed?

**Pull request merged**